plugins/net_plugin/net_plugin.cpp

```diff
@@ -3428,10 +3474,40 @@ namespace eosio {
       }
    }
 
+   // called from any thread
+   void net_plugin_impl::update_security_group(const block_state_ptr& bs) {
+      // update cache
+      //
+      auto& update = bs->get_security_group_info();
+      std::lock_guard<std::shared_mutex> connection_guard(connections_mtx);
```

Comment:
still need to add something like:
// NOTE: since this is the only thread that ever updates security_group.version it is safe to check this before taking the lock
if (security_group.version == update.version) {
   return;
}

just before taking the lock to avoid taking the lock on every block when we only need to when the version changed.